clang/include/clang/Basic/BuiltinsWebAssembly.def

```diff
@@ -192,6 +192,7 @@ TARGET_BUILTIN(__builtin_wasm_relaxed_dot_bf16x8_add_f32_f32x4, "V4fV8UsV8UsV4f"
 
 // Half-Precision (fp16)
 TARGET_BUILTIN(__builtin_wasm_loadf16_f32, "fh*", "nU", "half-precision")
+TARGET_BUILTIN(__builtin_wasm_storef16_f32, "vfh*", "nU", "half-precision")
```

Comment:
Can stores be considered pure?

Reply:
What does `pure` mean in this context? The docs in clang/Basic/Builtins.def don't have any info on this.

Reply:
The only documentation I know of for this is https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html
Clang has https://clang.llvm.org/docs/AttributeReference.html but it's not helpful here.

I guess a store shouldn't be pure, since it has an observable side effect, and in any case it doesn't return a value.

Reply:
I'm not exactly sure about the definition, but Builtins.def says it should be in sync with `Builtin::Context` class:
https://github.com/llvm/llvm-project/blob/b942c24845a39e6161c8623b1efc4e2083d879e9/clang/include/clang/Basic/Builtins.def#L67-L68

And `Builtin::Context` says this:
https://github.com/llvm/llvm-project/blob/b942c24845a39e6161c8623b1efc4e2083d879e9/clang/include/clang/Basic/Builtins.h#L115-L118

So I guess stores cannot be considered as pure, because they have side effects.